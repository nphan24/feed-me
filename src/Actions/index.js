export const postRecipes = recipes => ({
  type: 'POST_RECIPES',
  recipes
});

export const replaceRecipes = recipes => ({
  type: 'REPLACE_RECIPES',
  recipes
});

export const addFavorite = recipe => ({
  type: 'ADD_FAVORITE',
  recipe
});

export const removeFavorite = id => ({
  type: 'REMOVE_FAVORITE',
  id
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const promptSignIn = error => ({
  type: 'PROMPT_SIGNIN',
  error
});