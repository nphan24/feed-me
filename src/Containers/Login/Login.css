.login-component {
  background-color: #e2dcdc;
  width: 500px;
  margin: auto auto;
  text-align: center;
  box-shadow: -10px 5px 15px grey;
}

.input-field {
  display: block;
  margin: auto;
  width: 300px;
  height: 40px;
  margin-top: 15px;
  border-radius: 15px;
  padding-left: 15px;
  padding-top: 5px;
  font-size: 36px;
}

.input-field::-webkit-input-placeholder {
  font-size: 48px;
}

.bottom-field {
  margin-bottom: 15px;
}

.login-comment {
  margin: 0;
  padding-top: 50px;
  font-size: 2.5em;
}

.login-submit-button {
  margin-bottom: 15px;
  width: 120px;
  height: 40px;
  border-radius: 15px;
  font-size: 1em;
  color: black;
}

.login-submit-button:hover {
  background-color: #abd8ea;
}

.signin-route {
  font-size: 1.5em;
  margin-bottom: 30px;
  text-decoration: none;
}

.signin-route:hover {
  color:#BA55D3;
}

.login-error {
  font-size: 2em;
  color: red;
}



