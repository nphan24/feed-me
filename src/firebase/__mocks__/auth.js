export const login = jest.fn().mockImplementation(() => [{
  email: 'dog@gmail.com', 
  password: 'dogdog', 
  uid: 24}
]);