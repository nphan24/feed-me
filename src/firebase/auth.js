import { auth } from './firebase';

//sign up
export const signUp = (email, password) => 
  auth.createUserWithEmailAndPassword(email, password);

  //login
export const login = (email, password) =>
  auth.signInWithEmailAndPassword(email, password);

  //signout
export const doSignOut = () => auth.signOut();