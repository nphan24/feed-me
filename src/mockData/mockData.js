/* eslint-disable */

export const mockRecipe = {
  name: 'pasta',
  image: 'url',
  totalTime: '30',
  id: 2
};

export const uncleanedMockRecipeData = { matches: [
  { recipeName: 'tofu', 
    imageUrlsBySize: 'url', 
    ingredients: ['',''], 
    totalTimeInSeconds: 3600, 
    id: 2,
    pasta: 'speghetti',
    fruit: 'mango'
  }
]
 };

export const cleanedMockRecipeData = [{ 
  Ingredients: ['', ''], 
  id: 2, 
  image: undefined, 
  name: 'tofu', 
  totaltime: 60 
}];

export const mockRecipesArray = [
  {
    name: 'pasta',
    image: 'url',
    totalTime: '30',
    id: 2
  },
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];

export const mockRemainingArray = [
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];
