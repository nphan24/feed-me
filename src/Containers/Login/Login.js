import React, { Component } from 'react';
import * as routes from '../../constants/routes';
import { connect } from 'react-redux';
import { Route, NavLink, withRouter } from 'react-router-dom';
import { auth } from '../../firebase';
import * as Actions from '../../Actions';
import './Login.css';
export class Login extends Component {
  constructor() {
    super();
    this.state = { 
      email: '', 
      password: '',
      error: false 
    };
  }

  handleInput = event => {
    const { name, value } = event.target;

    this.setState({ [name]: value });
  };

  handleSubmit = async (event) => {
    event.preventDefault();
    const {email, password } = this.state;

    try {
      const authUser = await auth.login(email, password);
      const user = { 
        email: authUser.email,
        password: authUser.password,
        uid: authUser.uid 
      };
      this.props.addUser(user);
      this.setState ({
        email: '',
        password: '',
        error: false
      });
      this.props.history.push('/');
    } catch (error) {
      this.setState({ error });
    }
  };

  render() {
    const { email, password, error } = this.state;

    return (
      <div className="login-component">
        <p className="login-comment">Please Login Below</p>
        <form className="login-form" 
          onSubmit={this.handleSubmit}>
          <label>
            <input 
              type="text" 
              placeholder="email" 
              className="input-field" 
              value={email} 
              name="email" 
              onChange={this.handleInput} 
            />
            <input 
              type="password" 
              placeholder="Password" className="input-field bottom-field" 
              value={password} 
              name="password" 
              onChange={this.handleInput} 
            />
            <button className="login-submit-button">Login</button>
          </label>
        </form>
        <NavLink 
          to={routes.SIGN_UP} 
          className="signin-route">
          Don't have an account? Sign Up!
        </NavLink>
        {error && <p className="login-error">{error.message}</p>}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(Actions.addUser(user))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login));

