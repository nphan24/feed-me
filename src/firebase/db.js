import { db } from './firebase';

export const doCreateUser = (uid, username, email) => {
  db.ref(`users/${uid}`).set({
    username,
    email,
    uid
  });
};

export const onceGetUsers = () => 
  db.ref('users').once('value');

export const addFavoriteToDB = (uid, favorite) => {
  db.ref(`favorites/${uid}`).set(
    favorite
  );
};

export const removeFavoriteFromDB = async (uid, favoriteId) => {
  const userFavorites = await getFavorites(uid);
  delete userFavorites[favoriteId];
  addFavoriteToDB(uid, userFavorites);
};

export const getFavorites = (uid) => {
  const favorites = db.ref(`favorites/${uid}`).once('value');
  return favorites;
};