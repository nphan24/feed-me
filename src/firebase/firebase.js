import * as firebase from 'firebase';

const config = {
  apiKey: 'AIzaSyCaGyiQQKcQbzZgiCyHLLHIGOvX5E4VrH0',
  authDomain: 'feed-me-2be60.firebaseapp.com',
  databaseURL: 'https://feed-me-2be60.firebaseio.com',
  projectId: 'feed-me-2be60',
  storageBucket: 'feed-me-2be60.appspot.com',
  messagingSenderId: '713865205266'
};

firebase.initializeApp(config);

const auth = firebase.auth();
const database = firebase.database();

export { auth };
