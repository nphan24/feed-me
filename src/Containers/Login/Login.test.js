import React from 'react';
import {Login, mapStateToProps, mapDispatchToProps} from './Login';
import { shallow } from 'enzyme';
import * as mock from '../../mockData/mockData';
import * as Actions from '../../Actions';
import * as firebase from '../../firebase/auth';
import { auth } from '../../firebase';

describe('Login', () => {
  let wrapper;
  let mockaddUser = jest.fn();
  let mockUser = mock.mockUser;
  let mockHistory = { push: jest.fn() };

  beforeEach(() => {
    mockaddUser = jest.fn();

    wrapper = shallow (
      <Login 
        addUser = {mockaddUser}
        user = {mockUser}
        history= {mockHistory}
      />
    );
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on input change', () => {
    const mockEvent = { target: { value: 'dog@gmail', name: 'dog'}};
    wrapper.instance().handleInput(mockEvent);
    expect(wrapper.state('dog')).toEqual('dog@gmail');
  });

  it('should call login on handleSubmit', () => {
    const event = { preventDefault: jest.fn() };
    firebase.login = jest.fn();
    wrapper.instance().handleSubmit(event);
    expect(firebase.login).toHaveBeenCalled();
  });

  it('should call addUser on handleSubmit', () => {
    // const event = { preventDefault: jest.fn() };
    // wrapper.instance().handleSubmit(event);
    // expect(mockaddUser).toHaveBeenCalledWith(mockUser);
  }); 

  it('should set state back to its default state', () => {

  });

  it('should call history.push with the correct params', () => {
    // wrapper.instance().handleSubmit();
    // expect(mockHistory.push).toHaveBeenCalledWith('/');
  });

  it('should state the state to an error if the login to firebase fails', () => {

  });
});

describe('mapStateToProps', () => {
  it('correctly maps users to props ', () => {
    const user = mock.mockUser;
    const expected = user;
    const mockState = { user };
    const mapped = mapStateToProps(mockState);
    expect(mapped.user).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with correct params on addUser', () => {
    const mockDispatch = jest.fn();
    const user = mock.mockUser;
    const expected = Actions.addUser(user);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addUser(user);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});