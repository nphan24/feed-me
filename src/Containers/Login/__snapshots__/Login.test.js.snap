// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    addUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="login-comment"
>
          Please Login Below
</p>,
        <form
          className="login-form"
          onSubmit={[Function]}
>
          <label>
                    <input
                              className="input-field"
                              name="email"
                              onChange={[Function]}
                              placeholder="email"
                              type="text"
                              value=""
                    />
                    <input
                              className="input-field bottom-field"
                              name="password"
                              onChange={[Function]}
                              placeholder="Password"
                              type="password"
                              value=""
                    />
                    <button
                              className="login-submit-button"
                    >
                              Login
                    </button>
          </label>
</form>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="signin-route"
          to="/signup"
>
          Don't have an account? Sign Up!
</NavLink>,
        false,
      ],
      "className": "login-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please Login Below",
          "className": "login-comment",
        },
        "ref": null,
        "rendered": "Please Login Below",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label>
            <input
                        className="input-field"
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="text"
                        value=""
            />
            <input
                        className="input-field bottom-field"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <button
                        className="login-submit-button"
            >
                        Login
            </button>
</label>,
          "className": "login-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="input-field"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="text"
                value=""
/>,
              <input
                className="input-field bottom-field"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <button
                className="login-submit-button"
>
                Login
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field bottom-field",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "login-submit-button",
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "label",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Don't have an account? Sign Up!",
          "className": "signin-route",
          "to": "/signup",
        },
        "ref": null,
        "rendered": "Don't have an account? Sign Up!",
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="login-comment"
>
            Please Login Below
</p>,
          <form
            className="login-form"
            onSubmit={[Function]}
>
            <label>
                        <input
                                    className="input-field"
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="email"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="input-field bottom-field"
                                    name="password"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                                    value=""
                        />
                        <button
                                    className="login-submit-button"
                        >
                                    Login
                        </button>
            </label>
</form>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="signin-route"
            to="/signup"
>
            Don't have an account? Sign Up!
</NavLink>,
          false,
        ],
        "className": "login-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please Login Below",
            "className": "login-comment",
          },
          "ref": null,
          "rendered": "Please Login Below",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label>
              <input
                            className="input-field"
                            name="email"
                            onChange={[Function]}
                            placeholder="email"
                            type="text"
                            value=""
              />
              <input
                            className="input-field bottom-field"
                            name="password"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value=""
              />
              <button
                            className="login-submit-button"
              >
                            Login
              </button>
</label>,
            "className": "login-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="input-field"
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  type="text"
                  value=""
/>,
                <input
                  className="input-field bottom-field"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
/>,
                <button
                  className="login-submit-button"
>
                  Login
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-field",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "email",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-field bottom-field",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "login-submit-button",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
            ],
            "type": "label",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Don't have an account? Sign Up!",
            "className": "signin-route",
            "to": "/signup",
          },
          "ref": null,
          "rendered": "Don't have an account? Sign Up!",
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
